keps/sig-scheduling/4816-dra-prioritized-list/README.md

```diff
@@ -440,35 +838,25 @@ spec:
-request could be met by several different types of GPU, in the listed order of
-preference. Each of those is a separate `DeviceRequest`, with both a
-`deviceClassName` and also its own name. The fact that these subrequests also
-have their own names allows us to apply constraints or configuration to
-specific, individual subrequests, in the event that it is the chosen
-alternative. In this example, the "small-gpu" choice requires a configuration
-option that the other two choices do not need. Thus, if the resolution of the
-"gpu" request is made using the "small-gpu" subrequest, then that configuration
-will be attached to the allocation. Otherwise, it will not.
-
-Similarly, for `Constraints`, the list of requests can include the main request
-name ("gpu" in this case), in which case the constraint applies regardless of
-which alternative is chosen. Or, it can include the subrequest name, in which
-case that constraint only applies if that particular subrequest is chosen.
-
-In the PodSpec, however, the subrequest names are not valid. Only the main
-request name may be used.
+For both the `v1beta1` and `v1beta2` APIs, the `status.allocationResult.devices.results[].request`
+field will reference the name of the main request for requests without
+alternatives that set the `Exactly` field on the request, while it will
+reference the selected subrequest on the format `<main request>/<subrequest>`
+for requests that provide alternatives by setting the `FirstAvailable` field.
 
 ### Scheduler Implementation
 
 Currently, the scheduler loops through each entry in `DeviceClaim.Requests` and
-tries to satisfy each one. This would work essentially the same, except that
-today, it [throws an
-error](https://github.com/kubernetes/kubernetes/blob/03f134461462f86239067ec20ec17a0ba892db52/staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go#L164)
-when it encounters a claim with a missing `DeviceClassName`. Instead, here we
-would check for entries in `FirstAvailableOf`, and add an additional loop,
-trying each of these requests in order.
+tries to satisfy each one. This would work essentially the same, except that if
+the `FirstAvailable` object is set, it will attempt all alternatives for a 
+request until one is found or the list is exhausted.
+
+The current implementation in the v1beta1 API uses a missing `DeviceClassName` to make sure that
+an old version of the scheduler doesn't incorrectly handle a request that
+sets `FirstAvailable` which it doesn't see. If this happens it will result in an
+[error](https://github.com/kubernetes/kubernetes/blob/03f134461462f86239067ec20ec17a0ba892db52/staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go#L164).
+For the v1beta2 API, this will be handled by the scheduler being able to
+detect claims that hasn't set the `Exactly` object on a request.
 
 The current implementation will navigate a depth-first search of the devices,
```

Comment:
It sounds like there we no changes to scoring, is that correct?

Reply:
That is correct. I've updated the language in the description and in the code comments to make this clearer.